src/Broadway/EventStore/DBALEventStore.php

```diff
@@ -69,10 +72,41 @@ public function __construct(
     /**
      * {@inheritDoc}
      */
-    public function load($id)
+    public function load($id, $playhead = 0)
     {
         $statement = $this->prepareLoadStatement();
         $statement->bindValue('uuid', $this->convertIdentifierToStorageValue($id));
+        $statement->bindValue('playhead', $playhead);
+        $statement->execute();
+
+        $events = $this->executeStatement($statement);
+
+        if (empty($events)) {
+            throw new EventStreamNotFoundException(sprintf('EventStream not found for aggregate with id %s', $id));
+        }
```

Comment:
I don't really think the extra query is worth the trouble. How could there ever be no events when you load from a snapshot? Then you would have to created a snapshot for an aggregate without events (which in normal cases is already impossible), right?